libs/gltfio/src/AssetLoader.cpp

```diff
@@ -785,14 +789,16 @@ bool FAssetLoader::createPrimitive(const cgltf_primitive* inPrim, Primitive* out
     }
 
     if (needsDummyData) {
-        uint32_t size = sizeof(ubyte4) * vertexCount;
-        BufferObject* bufferObject = BufferObject::Builder().size(size).build(*mEngine);
-        mResult->mBufferObjects.push_back(bufferObject);
-        uint32_t* dummyData = (uint32_t*) malloc(size);
-        memset(dummyData, 0xff, size);
-        VertexBuffer::BufferDescriptor bd(dummyData, size, FREE_CALLBACK);
-        bufferObject->setBuffer(*mEngine, std::move(bd));
-        vertices->setBufferObjectAt(*mEngine, slot, bufferObject);
+        const uint32_t requiredSize = sizeof(ubyte4) * vertexCount;
+        if (mDummyBufferObject == nullptr || requiredSize > mDummyBufferObject->getByteCount()) {
+            mDummyBufferObject = BufferObject::Builder().size(requiredSize).build(*mEngine);
+            mResult->mBufferObjects.push_back(mDummyBufferObject);
+            uint32_t* dummyData = (uint32_t*) malloc(requiredSize);
+            memset(dummyData, 0xff, requiredSize);
```

Comment:
Is the memset necessary?

Reply:
Yes, ubershader handles per-vertex color, which needs to be set to 1.0